src/core/server/saved_objects/migrationsv2/model/retry_state.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { State } from '../types';
+
+export const delayRetryState = <S extends State>(
+  state: S,
+  errorMessage: string,
+  /** How many times to retry a step that fails */
+  maxRetryAttempts: number
```

Comment:
File extracted from `model.ts`

Reply:
Is there a test file for this helper?

Reply:
No, there was no unit tests for the part of the code I extracted to this file. Can add some.